packages/autocertifier-client/src/RestClient.ts

```diff
@@ -0,0 +1,108 @@
+import { Session } from './data/Session'
+import { CertifiedSubdomain } from './data/CertifiedSubdomain'
+import request from 'request'
+import { UpdateIpAndPortRequest } from './data/UpdateIpAndPortRequest'
+import { CreateCertifiedSubdomainRequest } from './data/CreateCertifiedSubdomainRequest'
+import { ServerError } from './errors'
+import { Logger } from '@streamr/utils'
+
+const logger = new Logger(module)
+
+// TODO: use a non-deprecated HTTP client that support async/await instead of request
```

Comment:
Possibly yes, the library has to support the option for setting `rejectUnauthorized: false`, this seems to be possible with fetch but needs testing. Alternatively we could just use legitimate TLS certificates on the server but this would make DDOS attacks from the browser possible which Petri was concerned about.